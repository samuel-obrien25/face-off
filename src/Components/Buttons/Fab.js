import React from 'react';
import '../Buttons/Buttons.css';

const Fab = (props) => {
  return (
    <button className={`fade-in ${props.fabClass}`}
            onClick={props.onClick}
            visibile={props.isActive}
    >   
        <h3>{props.fabText}</h3>
    </button>
  )
}

export default Fab;
