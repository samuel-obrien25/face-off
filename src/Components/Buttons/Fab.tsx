import React, {useState} from 'react';
import styled, {keyframes} from 'styled-components';
import '../../Styles/league-colors.css';

//#region STYLES
const StyledFab = styled.button`
  border-radius: 50%;
  border: none;
  box-shadow: 2px 2px 6px rgba(0,0,0,.4);
  transition: .5s ease-in-out;
  position: fixed;
  bottom: 15px;
  right: 15px;
  height: 75px;
  width: 75px;
    &:focus {
      outline: none;
    }
`;
const StyledHamburger = styled.div`
    width: 30px;
    height: 2px;
    margin: auto;
    background-color: currentColor;
    transition: .25s ease-in-out;

    ::before, ::after{
      content:'';
      width: 30px;
      height: 2px;
      background-color: inherit;
      position: absolute;
      left: 0;
      right: 0;
      margin: auto;
      transition: .2s ease-in-out;
    }

    ::before{
      top: 35%;
    }

    ::after{
      bottom: 35%;
    }
`;

const StyledFabH2 = styled.h2`
    margin: auto;
    color: currentColor;
    position: relative;
    text-align: center;
`;

const ActiveFabWrapper = styled.div`
    width: 100vw;
    height: 100vh;
    position: fixed;
    top: 0;
    left: 0;
    background-color: rgba(0,0,0,.4);
    z-index: 9999;
`;

const ActiveFabOne = styled(StyledFab)`

`;

const FabSlideUp = keyframes`
  from {
    opacity: 0;
    transform: translateY(125px);
  }
  to {
    opacity: 1;
    transform: translateY(0px);
  }
`

const ActiveFabTwo = styled(StyledFab)`
    animation: ${FabSlideUp} .45s cubic-bezier(0.175, 0.885, 0.32, 1.275) forwards;
    bottom: 115px;
`

const ActiveFabThree = styled(StyledFab)`
    animation: ${FabSlideUp} .6s cubic-bezier(0.175, 0.885, 0.32, 1.275) forwards;
    bottom: 215px;
`
//#endregion STYLES

interface FabProps {
  activeTeamID: String,
  handleClick: (event: React.MouseEvent<HTMLButtonElement>) => void,
  isActive: String,
}

const Fab: React.FC<FabProps> = (props) => {

  const [isActive, setIsActive] = useState<boolean>(false);
  
  const { activeTeamID, handleClick } = props;

  function scrollToNextGame() {
    
    /* Those exlamation marks are non-null assertion operators. Need to read more about them.
    * https://www.logicbig.com/tutorials/misc/typescript/non-null-assertion-operator.html
    */

    const lastGame = document.getElementById('card_container_schedule_list')!.lastElementChild;
    const nextGame = document.querySelector('[data-pastfuture="future"]')!;

    if(!document.querySelector('[data-pastfuture="future"]')){
      if(lastGame){
        lastGame.scrollIntoView({
          behavior: 'smooth',
          block: 'start'
        });
      }
      setIsActive(!isActive);
      return;
    } else {
      nextGame.scrollIntoView({
        behavior: 'smooth',
        block: 'start'
      });
      setIsActive(!isActive);
    }

  }

    if(isActive) {
      return (
        <ActiveFabWrapper>
          <ActiveFabThree className={`team${activeTeamID}`} onClick={scrollToNextGame}>
            <StyledFabH2>Next Game</StyledFabH2>
          </ActiveFabThree>

          <ActiveFabTwo className={`team${activeTeamID}`} onClick={handleClick}>
            <StyledFabH2>Teams</StyledFabH2>
          </ActiveFabTwo>

          <ActiveFabOne className={`team${activeTeamID}`} onClick={() => setIsActive(!isActive)}>
            <StyledFabH2>Close</StyledFabH2>
          </ActiveFabOne>
        </ActiveFabWrapper>
      );
    }

    return (
      <>
        <StyledFab className={`team${activeTeamID}`} onClick={() => setIsActive(!isActive)}>
          <StyledHamburger></StyledHamburger>
        </StyledFab>
      </>
    );
}

export default Fab;
