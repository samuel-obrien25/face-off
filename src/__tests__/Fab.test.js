import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Fab from '../Components/Buttons/Fab';

describe('Fab', () => {

    it('should be defined', () => {
        expect(Fab).toBeDefined();
    });

    it('Should render correctly', () => {
        const component = shallow(
            <Fab fabText="Fab Test" />
        );
        expect(component).toMatchSnapshot();
    });

    it('Should be a button', () => {
        const string = 'test string';
        const component = shallow(
            <Fab fabText={string} />
        );
        expect(component.type()).to.equal('button');
                 
    });
});