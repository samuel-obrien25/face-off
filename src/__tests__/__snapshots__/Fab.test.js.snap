// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Fab
    fabText="Fab Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3>
        Fab Test
      </h3>,
      "className": "fade-in undefined",
      "onClick": undefined,
      "visibile": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Fab Test",
      },
      "ref": null,
      "rendered": "Fab Test",
      "type": "h3",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          Fab Test
        </h3>,
        "className": "fade-in undefined",
        "onClick": undefined,
        "visibile": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fab Test",
        },
        "ref": null,
        "rendered": "Fab Test",
        "type": "h3",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
